### PR TITLE
Added some documentation on how to set up on citycloud + flavor

### PR DESCRIPTION
**Pull request by Xartos**
Added openstack/citycloud flavor and added some snippets I used to set it up with.

### CODE REVIEW

**Comment by cristiklein** [comment]
Can we add something similar in compliantkubernetes.io?

Ideally, it should only require copying the MD in the right place, updating
`mkdocs.yaml` and checking with `mkdocs serve`.

**Comment by Xartos** [comment]
> Can we add something similar in compliantkubernetes.io? Ideally, it should only require copying the MD in the right place, updating `mkdocs.yaml` and checking with `mkdocs serve`.

Yes, this was my end goal. But I think we might want to actually take some more time to polish this before we add it there. So I thought I just add this here for now so people that need to set it up have something to copy-paste. And then we can create an issue and polish it for compliantkubernetes.io

File: docs/citycloud-snippets.md
Comment (lentzi90) [inline docs/citycloud-snippets.md:35]: I know this is where the openstack terraform code/docs thinks it should be, but I think it would be better if we could stick to the `$CK8S_CONFIG_PATH`.  My suggestion is to link `contrib/terraform/openstack` into `$CK8S_CONFIG_PATH` or  `$CK8S_CONFIG_PATH/$CLUSTER-config`. This way we keep the terraform state and `cluster.tfvars` under the config path so it is easier to keep track of it.

File: bin/init.bash
Comment (lentzi90) [inline bin/init.bash:35]: Not sure about this, but let me know what you think. What about skipping citycloud here and just treating it as generic openstack? If we want something specifically for citycloud we can add it later.

**Review by lentzi90 — COMMENTED** [review]
Really nice job! It is interesting to see the different ways to do this. :smile: 

Added some comments below. Let me know what you think!

File: docs/citycloud-snippets.md
Comment (Xartos) [reply docs/citycloud-snippets.md:70]: Yea, I mainly followed the [AWS](https://compliantkubernetes.io/operator-manual/aws/) / [Exoscale](https://github.com/elastisys/compliantkubernetes/pull/41) guide for this code.
Maybe @cristiklein has some thought of this particular decision?

File: docs/citycloud-snippets.md
Comment (Xartos) [reply docs/citycloud-snippets.md:20]: Yes, I should add a short mentioning of how to authenticate. Maybe link to some documentation.

The reason is mainly because I wanted to, once again,  align with the aws / exoscale guide

File: bin/init.bash
Comment (Xartos) [reply bin/init.bash:35]: Yes I agree. I mostly wanted to get this here to show my thinking. But since it seems to be the same for safespring (for what we know now) I think we can change it to openstack

File: docs/citycloud-snippets.md
Comment (cristiklein) [reply docs/citycloud-snippets.md:35]: I prefer this approach. It is closer aligned with kubespray. Terraform state should eventually be handled with `-backend-config`.

File: docs/citycloud-snippets.md
Comment (cristiklein) [reply docs/citycloud-snippets.md:70]: Auto-approve was to simplify copy-paste-ablility. I'm find with eliminating `-auto-approve` as long as the code can be copy-pasted. Last time I checked, this required escaping newline, to make bash "wait" until the very end before starting to ask approve-related questions.

File: docs/citycloud-snippets.md
Comment (cristiklein) [inline docs/citycloud-snippets.md:94]: This is a thorn in my eye. Looking forward to getting rid of it. :smile:

File: docs/citycloud-snippets.md
Comment (cristiklein) [inline docs/citycloud-snippets.md:97]: I know what you mean here, but it feels weird to make `inventory.ini` executable. Making `hosts` executable feels more expected. Anyway, no need to fix this documentation, but please add an issue to `compliantkubernetes-kubespray` to handle both "static" inventory.ini and "dynamic" hosts.

File: docs/citycloud-snippets.md
Comment (cristiklein) [reply docs/citycloud-snippets.md:117]: Is that really that dramatic? At least in AWS, the resources created by Kubernetes are clearly tagged and can easily be removed manually. Maybe we should find the snipped that does so?

**Review by cristiklein — APPROVED** [review]

File: docs/citycloud-snippets.md
Comment (lentzi90) [reply docs/citycloud-snippets.md:70]: I see. It is probably fine to keep the auto approve, but I would suggest a note about it to make it more obvious to people. Otherwise I think it may surprise people that are used to always confirming terraform apply.

File: docs/citycloud-snippets.md
Comment (lentzi90) [reply docs/citycloud-snippets.md:117]: I don't think it is that easy to find them in other clouds but I could be wrong. But more importantly, this could cause a surprise bill for people who aren't aware of it. If you know terraform but not kubernetes, it wouldn't be obvious at all that there can be resources left in the cloud after you do `terraform destroy`.

Relevant issue: https://github.com/elastisys/compliantkubernetes-apps/issues/213

File: docs/citycloud-snippets.md
Comment (lentzi90) [reply docs/citycloud-snippets.md:35]: Ok the state can be handled with a backend config, but what about `cluster.tfvars`? Surely we should keep that in the config folder?

File: docs/citycloud-snippets.md
Comment (cristiklein) [reply docs/citycloud-snippets.md:117]: I see two aspects you bring here:

1. Regarding the Operator Manual, it must start from the assumption that people have CKA or equivalent under the bell. That curriculum does a way better job at explaining the ins-and-outs of Kubernetes than we will ever do. We should focus our documentation on what Compliant Kubernetes *adds* on top of a vanilla Kubernetes cluster, i.e., on what differentiates us, and not redo a half-finished, incomplete, out-of-date version of CKA. :smile: 
2. If the interaction between Kubernetes and underlying cloud resources is not well known or well understood in the community, this would be an excellent blog post.

File: docs/citycloud-snippets.md
Comment (cristiklein) [reply docs/citycloud-snippets.md:35]: Kubeconfig encourages keeping `cluster.tfvars` in the `inventory/$CLUSTER` folder.

Comment (lentzi90) [reply docs/citycloud-snippets.md:35]: Yes but I don't think it makes sense to have it outside of `$CK8S_CONFIG_PATH`. If we want it in the inventory to stick to the kubespray examples, then I think we should change the config path like this:

```bash
export CK8S_CONFIG_PATH="inventory/${CLUSTER}"
```

Comment (cristiklein) [reply docs/citycloud-snippets.md:35]: I see your point. Let's go with your suggestion.

File: docs/citycloud-snippets.md
Comment (Xartos) [reply docs/citycloud-snippets.md:10]: Not at all. But shellcheck complained about it so I changed it

File: docs/citycloud-snippets.md
Comment (Xartos) [reply docs/citycloud-snippets.md:97]: This is a requirement for kubespray. Since it couldn't execute this script without having permission to execute it

File: bin/init.bash
Comment (Xartos) [reply bin/init.bash:35]: You certainly have a point there

File: bin/init.bash
Comment (Xartos) [reply bin/init.bash:89]: Hmm, would that work. Since when we copy these files the structure changes so the links would need to change. 

If I understand you correctly the structure would look something like this:
```
openstack/group_vars/some_openstack_file.txt
citycloud/group_vars/some_citycloud_file.txt -> ../../openstack/group_vars/some_openstack_file.txt
```
And after you copied the citycloud config it would look like this:
```
group_vars/some_citycloud_file.txt -> ../../openstack/group_vars/some_openstack_file.txt
```
Or am I missing something?

File: docs/citycloud-snippets.md
Comment (cristiklein) [reply docs/citycloud-snippets.md:97]: I get that the script needs to be executable for kubespray. However, I would prefer that executable to be called `hosts` (without extension) and not `something.ini`. The `ini` part feels misleading.

File: bin/init.bash
Comment (cristiklein) [reply bin/init.bash:89]: Nice catch @Xartos ! I think you would need to use `cp --dereference`. I think this is already done some in other parts of kubespray.

Comment (Xartos) [reply bin/init.bash:89]: Right, yea then I agree. That is a quite neat solution

File: docs/citycloud-snippets.md
Comment (Xartos) [reply docs/citycloud-snippets.md:97]: Right, yes that is a bit misleading. And yes, creating a new issue for this is probably the best for now.

File: bin/init.bash
Comment (Xartos) [reply bin/init.bash:89]: Actually, I think that this change might go a bit too far for this issue. I'll create a new issue for this instead.

**Review by lentzi90 — APPROVED** [review]
LGTM

**Pull request merged**
